lgc/state/PalMetadata.cpp

```diff
@@ -447,11 +448,33 @@ void PalMetadata::finalizePipeline() {
   paClClipCntl.bits.DX_RASTERIZATION_KILL = rasterizerDiscardEnable;
   setRegister(mmPA_CL_CLIP_CNTL, paClClipCntl.u32All);
 
+  if (m_pipelineState->isGraphics()) {
+    if (m_registers.find(m_document->getNode(mmSPI_SHADER_COL_FORMAT)) == m_registers.end()) {
+      llvm::SmallVector<ColorExportInfo, 8> exp;
+      getColorExportInfo(exp);
+      updateSpiShaderColFormat(exp);
```

Comment:
I'm not sure about doing this here. I was thinking that there would be code in FragColorExport.cpp that adds the export instructions and calls the function to add the PAL metadata. That code would be run as part of the new pass for the pipeline compilation case, and would be run from the glue shader generation in the unlinked case. Would that work?

Reply:
> would be run from the glue shader generation in the unlinked case.

You're right.  I was thinking about this all wrong.  For some reason I had in my head that an update to the pal metadata would not be reflected in the elf blob, but it is.  The update will happen correctly when merging the pal metadata.